neo/rawio/axonarawio.py

```diff
@@ -81,57 +98,149 @@ def _parse_header(self):
+        params = {'bin': {'filename': self.bin_file,
+                          'bytes_packet': 432,
+                          'bytes_data': 384,
+                          'bytes_head': 32,
+                          'bytes_tail': 16,
+                          'data_type': 'int16',
+                          'header_size': 0,
+                          # bin files don't contain a file header
+                          'header_encoding': None},
+                  'set': {'filename': self.set_file,
+                          'header_encoding': 'cp1252'},
+                  'unit': {'data_type': unit_dtype,
+                           'tetrode_ids': [],
+                           'header_encoding': 'cp1252'}}
+        self.file_parameters = params
+
+        # SCAN SET FILE
+        set_dict = self.get_header_parameters(self.set_file, 'set')
+        params['set']['file_header'] = set_dict
+        params['set']['sampling_rate'] = int(set_dict['rawRate'])
+
+        # SCAN BIN FILE
+        signal_streams = []
+        signal_channels = []
+        if self.bin_file:
+            bin_dict = self.file_parameters['bin']
+            # add derived parameters from bin file
+            bin_dict['num_channels'] = len(self.get_active_tetrode()) * 4
+            num_tot_packets = int(self.bin_file.stat().st_size
+                                  / bin_dict['bytes_packet'])
+            bin_dict['num_total_packets'] = num_tot_packets
+            bin_dict['num_total_samples'] = num_tot_packets * 3
+
+            # Create np.memmap to .bin file
+            self._raw_signals = np.memmap(
+                self.bin_file, dtype=self.file_parameters['bin']['data_type'],
+                mode='r', offset=self.file_parameters['bin']['header_size']
+            )
+
+            signal_streams = self._get_signal_streams_header()
+            signal_channels = self._get_signal_chan_header()
+
+        # SCAN TETRODE FILES
+        # In this IO one tetrode corresponds to one unit
+        self._raw_spikes = []
+        spike_channels = []
+        if self.tetrode_files:
+
+            for i, tetrode_file in enumerate(self.tetrode_files):
+                # collecting tetrode specific parameters and dtype conversions
+                tdict = self.get_header_parameters(tetrode_file, 'unit')
+                tdict['filename'] = tetrode_file
+                tdict['num_chans'] = int(tdict['num_chans'])
+                tdict['num_spikes'] = int(tdict['num_spikes'])
+                tdict['header_size'] = len(
+                    self.get_header_bstring(tetrode_file))
+
+                # memory mapping spiking data
+                spikes = np.memmap(
+                    tetrode_file,
+                    dtype=self.file_parameters['unit']['data_type'],
+                    mode='r', offset=tdict['header_size'],
+                    shape=(tdict['num_spikes'] * 4))
+                self._raw_spikes.append(spikes)
+
```

Comment:
This dtype for spikes is amazing, do you mean that the spike time is duplicated on the 4 channels ? !!!

Reply:
Yes.

Reply:
changed to 2D implementation